lib/datasets/communities.rb

```diff
@@ -0,0 +1,184 @@
+      :percent_empl_prof_serv,
+      :percent_occup_manu,
+      :percent_occup_mgmt_prof,
+      :male_percent_divorce,
+      :male_percent_nev_marr,
+      :female_percent_div,
+      :total_percent_div,
+      :pers_per_fam,
+      :percent_fam2_par,
+      :percent_kids2_par,
+      :percent_young_kids2_par,
+      :percent_teen2_par,
+      :percent_work_mom_young_kids,
+      :percent_work_mom,
+      :num_illeg,
+      :percent_illeg,
+      :num_immig,
+      :percent_immig_recent,
+      :percent_immig_rec5,
+      :percent_immig_rec8,
+      :percent_immig_rec10,
+      :percent_recent_immig,
+      :percent_rec_immig5,
+      :percent_rec_immig8,
+      :percent_rec_immig10,
+      :percent_speak_engl_only,
+      :percent_not_speak_engl_well,
+      :percent_larg_house_fam,
+      :percent_larg_house_occup,
+      :pers_per_occup_hous,
+      :pers_per_own_occ_hous,
+      :pers_per_rent_occ_hous,
+      :percent_pers_own_occup,
+      :percent_pers_dense_hous,
+      :percent_hous_less3_br,
+      :med_num_br,
+      :hous_vacant,
+      :percent_hous_occup,
+      :percent_hous_own_occ,
+      :percent_vacant_boarded,
+      :percent_vac_more6_mos,
+      :med_yr_hous_built,
+      :percent_hous_no_phone,
+      :percent_wo_full_plumb,
+      :own_occ_low_quart,
+      :own_occ_med_val,
+      :own_occ_hi_quart,
+      :rent_low_q,
+      :rent_median,
+      :rent_high_q,
+      :med_rent,
+      :med_rent_percent_hous_income,
+      :med_own_cost_percent_income,
+      :med_own_cost_percent_income_no_mtg,
+      :num_in_shelters,
+      :num_street,
+      :percent_foreign_born,
+      :percent_born_same_state,
+      :percent_same_house85,
+      :percent_same_city85,
+      :percent_same_state85,
+      :lemas_sworn_ft,
+      :lemas_sw_ft_per_pop,
+      :lemas_sw_ft_field_ops,
+      :lemas_sw_ft_field_per_pop,
+      :lemas_total_req,
+      :lemas_tot_req_per_pop,
+      :polic_req_per_offic,
+      :polic_per_pop,
+      :racial_match_comm_pol,
+      :percent_polic_white,
+      :percent_polic_black,
+      :percent_polic_hisp,
+      :percent_polic_asian,
+      :percent_polic_minor,
+      :offic_assgn_drug_units,
+      :num_kinds_drugs_seiz,
+      :polic_ave_ot_worked,
+      :land_area,
+      :pop_dens,
+      :percent_use_pub_trans,
+      :polic_cars,
+      :polic_oper_budg,
+      :lemas_percent_polic_on_patr,
+      :lemas_gang_unit_deploy,
+      :lemas_percent_offic_drug_un,
+      :polic_budg_per_pop,
+      :violent_crimes_per_pop
+    )
+
+    def initialize
+      super()
+      @metadata.id = "communities"
+      @metadata.name = "Communities"
+      @metadata.url = "http://archive.ics.uci.edu/ml/datasets/communities+and+crime"
+      @metadata.description = lambda do
+        read_names
+      end
+    end
+
+    def each
+      return to_enum(__method__) unless block_given?
+
+      open_data do |csv|
+        csv.each do |row|
+          next if row[0].nil?
+          record = Record.new(*row)
```

Comment:
How about processing "?" by ourselves instead of `CSV.open` options?

```ruby
row = row.collect.with_index do |column, i|
  if column == "?"
    nil
  else
    column = Float(column) unless i == 3
    column
  end
end
record = Record.new(*row)
```